pkg/k8s/api.go

```diff
@@ -155,6 +155,15 @@ func (kubeAPI *KubernetesAPI) GetPodsByNamespace(ctx context.Context, namespace
 	return podList.Items, nil
 }
 
+// GetPods returns all pods in a given namespace, with the label selector
+func (kubeAPI *KubernetesAPI) GetPods(ctx context.Context, namespace string, selector string) ([]corev1.Pod, error) {
```

Comment:
Is calling

```
pods, err := kubeAPI.GetPods(ctx, ns, "foo=bar")
```

enough more ergonomic than

```
pods, err := kubeAPI.CoreV1().Pods(ns).List(ctx, metav1.ListOptions{LabelSelector: "foo=bar"})
```

to justify a layer of indirection?  I like the second one better because I can see directly what it does instead of having to look up the GetPods function to see what it does.

Reply:
Makes sense. My reasoning was that it's a common function and could be used at multiple places,,